pkg/registry/generic/registry/store.go

```diff
@@ -954,3 +956,110 @@ func (e *Store) Export(ctx api.Context, name string, opts metav1.ExportOptions)
 	}
 	return obj, nil
 }
+
+// CompleteWithOptions updates the store with the provided options and defaults common fields
+func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
+	if e.QualifiedResource.Empty() {
+		return fmt.Errorf("store %#v must have a non-empty qualified resource", e)
+	}
+	if e.NewFunc == nil {
+		return fmt.Errorf("store for %s must have NewFunc set", e.QualifiedResource.String())
+	}
+	if e.NewListFunc == nil {
+		return fmt.Errorf("store for %s must have NewListFunc set", e.QualifiedResource.String())
+	}
+
+	var isNamespaced bool
+	switch {
+	case e.CreateStrategy != nil:
+		isNamespaced = e.CreateStrategy.NamespaceScoped()
+	case e.UpdateStrategy != nil:
+		isNamespaced = e.UpdateStrategy.NamespaceScoped()
+	default:
+		return fmt.Errorf("store for %s must have CreateStrategy or UpdateStrategy set", e.QualifiedResource.String())
+	}
+
+	if options.RESTOptions == nil {
+		return fmt.Errorf("options for %s must have RESTOptions set", e.QualifiedResource.String())
+	}
+	if options.AttrFunc == nil {
+		return fmt.Errorf("options for %s must have AttrFunc set", e.QualifiedResource.String())
+	}
+
+	opts, err := options.RESTOptions.GetRESTOptions(e.QualifiedResource)
+	if err != nil {
+		return err
+	}
+
+	// Resource prefix must come from the underlying factory
+	prefix := opts.ResourcePrefix
+	if !strings.HasPrefix(prefix, "/") {
+		prefix = "/" + prefix
+	}
+
+	// Set the default behavior for storage key generation
+	if isNamespaced {
+		if e.KeyRootFunc == nil {
```

Comment:
I think if someone wants to customize keyfuncs, they should customize both... check at the top and err in this case:
```
if (e.KeyRootFunc == nil) != (e.KeyFunc == nil) {
...
}
```